lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -224,26 +227,37 @@ public Dictionary(
     this.needsInputCleaning = ignoreCase;
     this.needsOutputCleaning = false; // set if we have an OCONV
 
-    Path tempPath = getDefaultTempDir(); // TODO: make this configurable?
-    Path aff = Files.createTempFile(tempPath, "affix", "aff");
-
-    BufferedInputStream aff1 = null;
-    InputStream aff2 = null;
-    boolean success = false;
-    try {
-      // Copy contents of the affix stream to a temp file.
-      try (OutputStream os = Files.newOutputStream(aff)) {
-        affix.transferTo(os);
+    try (BufferedInputStream affixStream =
+        new BufferedInputStream(affix, MAX_PROLOGUE_SCAN_WINDOW) {
+          @Override
+          public void close() throws IOException {
+            // TODO: maybe we should consume and close it? Why does it need to stay open?
```

Comment:
Probably so that the callers who opened the streams can close them safely using try-with-resources.

Reply:
Closeable.close() can be invoked any number of times without side-effects - this is a contract stated in the javadoc.

Reply:
True. It feels a bit more right to me when the ones creating the streams close them. But in fact I don't like the whole idea of passing streams to constructor. I believe most clients would be happier with passing paths (except the rare(?) cases when the content is created in memory or loaded from classpath).

Reply:
Leave it a stream, it's hard to beat its flexibility.